me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeWebVerify.java

```diff
@@ -204,34 +200,49 @@ private String ExtractAccessToken(String url)
      */
     private void GetWebProfile(String url)
     {
+        String serverResponse;
+        HttpURLConnection urlConnection;
+        URL urlRequest;
+        try {
+            urlRequest = new URL(url);
+            urlConnection = (HttpURLConnection) urlRequest.openConnection();
+
+            int responseCode = urlConnection.getResponseCode();
+
+            if(responseCode == HttpURLConnection.HTTP_OK){
+                serverResponse = readStream(urlConnection.getInputStream());
+                SendDataBack(serverResponse);
+            }
+        } catch (IOException e) {
+            SendErrorBack(e.getMessage());
+        }
+    }
 
-        try
-        {
-            HttpClient httpclient = new DefaultHttpClient();
-            HttpResponse response = httpclient.execute(new HttpGet(url));
-            StatusLine statusLine = response.getStatusLine();
-
-            if (statusLine.getStatusCode() == HttpStatus.SC_OK)
-            {
-                ByteArrayOutputStream out = new ByteArrayOutputStream();
-                response.getEntity().writeTo(out);
-                String responseString = out.toString();
-
-                out.close();
-                SendDataBack(responseString);
-
+    /**
+     * This converts the InputStream to a String
+     *
+     * @param inputStream from the Web Request
+     * @return the converted string
+     */
+    private String readStream(InputStream inputStream) throws IOException {
+        BufferedReader reader = null;
+        StringBuilder response = new StringBuilder();
+        try {
+            reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
no, it's a java 7 feature, and we support android sdk 14, which works with java 6